wolfcrypt/src/logging.c

```diff
@@ -41,13 +41,31 @@
     } 
 #endif
 
+#if defined(OPENSSL_EXTRA) || defined(DEBUG_WOLFSSL_VERBOSE)
+static wolfSSL_Mutex debug_mutex; /* mutex for access to debug structure */
+
+/* accessing any of these global variables should be wrapped in a lock of
+ * debug_mutex */
+volatile char          wc_last_error_file[WOLFSSL_MAX_ERROR_SZ];
```

Comment:
This was cached for wolfSSL_ERR_peek_last_error_line, after having added the queue structure I think this can be simplified and just inspect the most recent node in the queue though.

Multithreaded access is handled by locking and unlocking debug_mutex before accessing the queue. Initializing debug_mutex was added to wolfCrypt_Init->wc_LoggingInit and clean up of it added to wolfCrypt_Cleanup->wc_LoggingCleanup.